tests/test_authorisation.py

```diff
@@ -321,6 +321,123 @@ def test_different_user_levels_see_correct_comments_on_sub_criteria_view(
+            - test commenter can not see comment set by assessor
+            - test commenter can not see comment set by lead assessor
+            - test commenter can see comment set by commenter
+            - test assessor can see all comments
+            - test lead assessor can see all comments
+        Args:
+            flask_test_client:
+
+        Returns:
+
+        """
+        flask_test_client.set_cookie(
+            "localhost",
+            "fsd_user_token",
+            create_valid_token(claim),
+        )
+
+        # Send a request to the route you want to test
+        application_id = request.node.get_closest_marker(
+            "application_id"
+        ).args[0]
+        sub_criteria_id = request.node.get_closest_marker(
+            "sub_criteria_id"
+        ).args[0]
+
+        mocker.patch(
+            "app.blueprints.assessments.routes.get_comments",
+            return_value=[
+                {
+                    "id": "test_id_1",
+                    "user_id": claim["accountId"],
+                    "date_created": "2022-12-08T08:00:01.748170",
+                    "theme_id": "test_theme_id",
+                    "sub_criteria_id": sub_criteria_id,
+                    "application_id": application_id,
+                    "updates": [
+                        {
+                            "comment": "This is old comment",
+                            "comment_id": "test_id_1",
+                            "date_created": "2022-12-08T08:00:01.748170",
+                        },
+                        {
+                            "comment": "This is comment has history",
+                            "comment_id": "test_id_1",
+                            "date_created": "2022-12-09T08:00:01.748170",
+                        },
+                    ],
+                },
+                {
+                    "id": "test_id_2",
+                    "user_id": claim["accountId"],
+                    "date_created": "2022-10-27T08:00:02.748170",
+                    "theme_id": "test_theme_id",
+                    "sub_criteria_id": sub_criteria_id,
+                    "application_id": application_id,
+                    "updates": [
+                        {
+                            "comment": "This is comment has no history",
+                            "comment_id": "test_id_2",
+                            "date_created": "2022-10-27T08:00:02.748170",
+                        }
+                    ],
+                },
+            ],
+        ),
+
+        response = flask_test_client.get(
+            f"/assess/application_id/{application_id}/sub_criteria_id/{sub_criteria_id}",  # noqa
+            follow_redirects=True,
+        )
+        assert response.status_code == 200
+
+        soup = BeautifulSoup(response.data, "html.parser")
+        all_comments = soup.find_all("div", class_="comment-group")
+
+        for comment in all_comments:
```

Comment:
Should there be an extra test to see the history of the comment (i.e. get the see history page and check it contains those two comments

Reply:
Good shout. Will do it

Reply:
Added the tests now